### PR TITLE
[Actions] Add action that will bump the hash used for the unified pipeline.

### PR DESCRIPTION
**Pull request by mandel-macaque**
tldr; The action will send an event to the sdk-insertions repository that will do the correct steps to bump the version of the android project to be used in the unified pipeline.

## Details

The idea is as follows:

The action will get the information of an update branch, that is sent to the sdk-insertions repo using a client payload via a repository dispatch (https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#repository_dispatch). That event with the payload is the parsed by a second action in the sdk-insertions repo and the needed yaml files are updated. This allow to have the unified pipeline build the latest commits of the SDKs without a manual trigger.

More details can be found here: https://github.com/xamarin/sdk-insertions/wiki/Unified-pipeline#github-actions

Because there is no way to run an action without a merge, please refer to the exact same one in the xamarin-macios repo: https://github.com/xamarin/xamarin-macios/actions/runs/3941098858

### CODE REVIEW

File: .github/workflows/sdk-insertion-bump.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+name: Notify release branch change
+
+on:
+  # trigger for main and release branches.
+  push:
+    branches:
+      - '*'
```
Comment (jonathanpeppers) [inline .github/workflows/sdk-insertion-bump.yml:7]: @jonpryor @pjcollins are GitHub Actions enabled on this repo? I was thinking we should see this new job run on the latest commit here:

https://github.com/xamarin/xamarin-android/tree/dev/mandel/sdk-insertions-bump

Just to know if this action will work or not.

Comment (mandel-macaque) [reply .github/workflows/sdk-insertion-bump.yml:7]: AFAIK it won't. I updated the commit message with a pointer to an example in the macios repo.

Comment (pjcollins) [reply .github/workflows/sdk-insertion-bump.yml:7]: I think we will want to change this branch filter to include `main` and `release/*` instead of a full wildcard if possible.

I am not sure if we need to do anything special to enable actions or to get this to run, it's probably ok to merge this and test it live.

**Review by pjcollins — APPROVED** [review]

**Review by jonathanpeppers — APPROVED** [review]

File: .github/workflows/sdk-insertion-bump.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+name: Notify release branch change
+
+on:
+  # trigger for main and release branches.
+  push:
+    branches:
+      - main
+      - 'release/**'
```
Comment (mandel-macaque) [inline .github/workflows/sdk-insertion-bump.yml:8]: @pjcollins  thans for this change

**Comment by jonpryor** [comment]
```markdown
Context: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions
Context: https://github.com/xamarin/sdk-insertions/wiki/Unified-pipeline#github-actions

Add a GitHub Actions Workflow in
`.github/workflows/sdk-insertion-bump.yml` which is run when new
commits are made to the `main` and `release/**` branches.

The GitHub Action will get the information of the updated branch and
send that commit informtaion to the xamarin/sdk-insertions repo using
a client payload via [`repository_dispatch`][0].  The event with the
payload is the parsed by
[xamarin/sdk-insertions/.github/workflows/sdk-bump.yml][1], which
updates various YAML files for the [unified insetion pipeline][2].
This allow to have the unified pipeline build the latest commits of
the SDKs without a manual trigger.

Co-authored-by: Peter Collins <pecolli@microsoft.com>

[0]: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#repository_dispatch
[1]: https://github.com/xamarin/sdk-insertions/blob/85976c35520b21cbde62a9412c8525ac2a5f4b84/.github/workflows/sdk-bump.yml
[2]: https://github.com/xamarin/sdk-insertions/wiki/Unified-pipeline#requirements
```

**Comment by pjcollins** [comment]
Looks like this worked https://github.com/xamarin/sdk-insertions/commit/731401bcf38fa984c1cb3576cfd85d9eceee9030
